fv3core/.jenkins/actions/fetch_caches.sh

```diff
@@ -2,24 +2,23 @@
 BACKEND=$1
 EXPNAME=$2
 SANITIZED_BACKEND=`echo $BACKEND | sed 's/:/_/g'` #sanitize the backend from any ':'
-CACHE_DIR="/scratch/snx3000/olifu/jenkins/scratch/store_gt_caches/"
+CACHE_DIR="/scratch/snx3000/olifu/jenkins/scratch/store_gt_caches/${EXPNAME}/${SANITIZED_BACKEND}"
```

Comment:
Here I updated the caching to use a tar.gz that has the gt4py version in its file path, so that we can have simultaneous caches for different gt4py versions. This was needed to be able to test this PR (which involves generating the caches) without breaking other PRs.